src/pages/LoginPage.jsx

```diff
@@ -1,18 +1,44 @@
 import TextField from "@mui/material/TextField";
 import Button from "@mui/material/Button";
 import Typography from "@mui/material/Typography";
+import { useNavigate } from "react-router-dom";
 import icons from "../assets/svg/svg-icons.svg";
 import "../assets/styles/index.css";
 import "@fontsource/roboto/300.css";
+import { useEffect, useState } from "react";
 export const LoginPage = () => {
+  const [formState, setformState] = useState({
+    email: "",
+    password: "",
+  });
+  const navigate = useNavigate();
+  const { email, password } = formState;
+  const onInputChange = ({ target }) => {
+    const { name, value } = target;
+    setformState({
+      ...formState,
+      [name]: value,
+    });
+  };
+  useEffect(() => {}, [formState]);
```

Comment:
This is unnecessary. 
You use the use effect when you want to trigger one action and the trigger will be wen the variables inside the array changes its value.